src/PDPVerifier.sol

```diff
@@ -445,12 +445,14 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         require(storageProvider[setId] == msg.sender, "Only the storage provider can add pieces");
         uint256 firstAdded = nextPieceId[setId];
         uint256[] memory pieceIds = new uint256[](pieceData.length);
+        bytes[] memory pieceCidsAdded = new bytes[](pieceData.length);
 
         for (uint256 i = 0; i < nPieces; i++) {
             addOnePiece(setId, i, pieceData[i]);
             pieceIds[i] = firstAdded + i;
+            pieceCidsAdded[i] = pieceCids[setId][firstAdded + i].data;
```

Comment:
it is this line that is triggering the compiler error

Reply:
Commenting out the `optimizer_runs` settings prevents the error

Reply:
How is the `pieceCidsAdded` array different from `pieceData`? Can the event payload use `pieceData` instead?